### PR TITLE
Added nf-test for goat/taxonsearch

### PR DESCRIPTION
**Pull request by GallVp**
Added nf-test for goat/taxonsearch with #6226

### CODE REVIEW

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    tag "goat/taxonsearch"
+
+    test("test-goat-taxonsearch-single-species") {
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test_single_species' ], // meta map
+				    taxon = 'Meles meles',
```
Comment (muffato) [inline modules/nf-core/goat/taxonsearch/tests/main.nf.test:21]: Can we remove the four `taxon = ` while we're at it ? They feel unnecessary 
```suggestion
				    'Meles meles',
```

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    test("test-goat-taxonsearch-genus-id") {
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test_genus_id' ], // meta map
+				    taxon = '7215',
```
Comment (muffato) [inline modules/nf-core/goat/taxonsearch/tests/main.nf.test:44]: Integers should be supported
```suggestion
				    7215,
```

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    test("test-goat-taxonsearch-species") {
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test_species' ], // meta map
+				    taxon = '',
```
Comment (muffato) [inline modules/nf-core/goat/taxonsearch/tests/main.nf.test:67]:
```suggestion
				    '',
```

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    test("test-goat-taxonsearch-species-stub") {
+        options '-stub'
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test_species' ], // meta map
+				    taxon = '',
```
Comment (muffato) [inline modules/nf-core/goat/taxonsearch/tests/main.nf.test:91]:
```suggestion
				    '',
```

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+				    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/goat_taxonsearch/taxonomy_ids.txt', checkIfExists: true)
```
Comment (muffato) [inline modules/nf-core/goat/taxonsearch/tests/main.nf.test:92]:
```suggestion
				    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/generic/txt/species_names.txt', checkIfExists: true)
```

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    test("test-goat-taxonsearch-species") {
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test_species' ], // meta map
+				    taxon = '',
+				    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/goat_taxonsearch/taxonomy_ids.txt', checkIfExists: true)
```
Comment (muffato) [inline modules/nf-core/goat/taxonsearch/tests/main.nf.test:68]:
```suggestion
				    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/generic/txt/species_names.txt', checkIfExists: true)
```

**Review by muffato — COMMENTED** [review]
The `delete_me` paths should be addressed now as well, to help https://github.com/nf-core/test-datasets/issues/1245
@GallVp : can you review https://github.com/nf-core/test-datasets/pull/1311 and we'll check that the CI here doesn't break

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test (hunk 5 above)
Comment (muffato) [reply modules/nf-core/goat/taxonsearch/tests/main.nf.test:92]: Should the path rather be changed to 
```suggestion
				    file(params.modules_testdata_base_path + 'generic/txt/species_names.txt', checkIfExists: true)
```

File: modules/nf-core/goat/taxonsearch/tests/main.nf.test (hunk 6 above)
Comment (muffato) [reply modules/nf-core/goat/taxonsearch/tests/main.nf.test:68]: Should the path rather be changed to 
```suggestion
				    file(params.modules_testdata_base_path + 'generic/txt/species_names.txt', checkIfExists: true)
```

**Comment by GallVp** [comment]
Thank you @muffato 

I have added the suggestions.

**Review by muffato — APPROVED** [review]
And all the tests pass. Youhou 🎉 !

**Pull request merged**